src/lib/consensus/proof_of_stake.ml

```diff
@@ -239,41 +241,67 @@ module Data = struct
           Table.add_exn last_checked_slot_and_epoch ~key:pk ~data ) ;
       last_checked_slot_and_epoch
 
-    let create block_producer_pubkeys ~genesis_ledger =
+    let create_epoch_ledger ~location ~genesis_ledger =
```

Comment:
In the genesis ledger case, this could cause a significant pause at bootup. And we can skip doing this if we just add some smart rules that redirect access of epoch ledgers where the hash is equal to the genesis ledger hash back to the original read-only genesis ledger database the daemon ships with. Don't fix this right now, but let's log an issue to do this work in the near future so that there isn't a pause when starting the node for the first time.